libs/langchain/langchain/output_parsers/json.py

```diff
@@ -101,10 +168,54 @@ class SimpleJsonOutputParser(BaseOutputParser[Any]):
     def parse(self, text: str) -> Any:
         text = text.strip()
         try:
-            return json.loads(text)
+            return parse_partial_json(text)
         except JSONDecodeError as e:
             raise OutputParserException(f"Invalid json output: {text}") from e
 
     @property
     def _type(self) -> str:
         return "simple_json_output_parser"
+
+
+class PartialFunctionsJsonOutputParser(BaseCumulativeTransformOutputParser[Any]):
+    @property
+    def _type(self) -> str:
+        return "partial_functions_json"
+
+    def parse_result(self, result: List[Generation]) -> Any:
+        if len(result) != 1:
+            raise OutputParserException(
+                f"Expected exactly one result, but got {len(result)}"
+            )
+        generation = result[0]
+        if not isinstance(generation, ChatGeneration):
+            raise OutputParserException(
+                "This output parser can only be used with a chat generation."
+            )
+        message = generation.message
+        try:
+            function_call = message.additional_kwargs["function_call"]
+        except KeyError:
+            return None
+        try:
+            return parse_partial_json(function_call["arguments"])
+        except KeyError:
+            return None
+
+    def _diff(self, prev: Optional[Any], next: Any) -> Any:
+        return jsonpatch.make_patch(prev, next).patch
+
+    def parse(self, text: str) -> Any:
```

Comment:
Could you leave a comment here to explain why this is not implemented? Could we raise an Exception instead to make sure that this code remains unreachable?